signer/src/keypair.rs

```diff
@@ -0,0 +1,138 @@
+//! Keypair structures and algorithms
+//!
+//! Definition of secret key, keypairs and related helpers
+
+use core::fmt;
+
+use crate::{CurvePoint, FieldHelpers, PubKey, ScalarField, SecKey};
+use ark_ec::{AffineCurve, ProjectiveCurve};
+use ark_ff::UniformRand;
+use rand::{self, CryptoRng, RngCore};
+
+/// Keypair structure
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub struct Keypair {
+    /// Secret key
+    pub secret: SecKey,
```

Comment:
any reason this is `pub`?

Reply:
Code in schnorr.rs needs access to secret.

Reply:
I'll `pub(crate)` it

Reply:
Ideally this should not be public IMO. Two things you could do:

* make it public only to the crate with `pub(crate)`, but you can still write code that will extract the secret key within the crate
* make it public only to this file, by removing `pub` and implementing signing on `Keypair`.